src/abstracts.ts

```diff
@@ -0,0 +1,9 @@
+export interface Queryable {
+	query(selector: string);
+}
+
+export interface ScratchProjectOptions {
+	file?: string,
+	uri?: string,
+	cloudId?: number
+}
```

Comment:
I think there is a way to simplify this API - I'm curious if the API was `type`/ `value`, I think it's weird you can pass in all these options to the same source, it complicates the code a bit (ex all your validation logic here https://github.com/bocoup/sb-util/pull/2/files#diff-2ada5d944dbe999424cb0fc42468c70cR23)

Can there be 3 paths? Something like initByFile, initById, etc..? Thoughts?

Reply:
Agree. Will implement this